session/macaroon.go

```diff
@@ -127,3 +128,26 @@ func RootKeyIDFromMacaroon(mac *macaroon.Macaroon) (uint64, error) {
 	// number.
 	return strconv.ParseUint(string(decodedID.StorageId), 10, 64)
 }
+
+// NewSessionPrivKeyAndID randomly derives a new private key and session ID
+// pair.
+func NewSessionPrivKeyAndID() (*btcec.PrivateKey, ID, error) {
+	var id ID
+
+	privateKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		return nil, id, fmt.Errorf("error deriving private key: %v",
+			err)
+	}
+
+	pubKey := privateKey.PubKey()
+
+	// NOTE: we use bytes [1:4] of the serialised public key to create the
+	// macaroon root key base along with the Session ID. This will provide
+	// 4 bytes of entropy. Previously, bytes [0:3] where used but this
+	// resulted in lower entropy due to the first byte always being either
+	// 0x02 or 0x03.
+	copy(id[:], pubKey.SerializeCompressed()[1:4])
```

Comment:
I'm pretty sure this needs to be `pubKey.SerializeCompressed()[1:5])`, otherwise we only get 3 bytes. The second index is _exclusive_. So the comment should be updated to read `Previously, bytes [0:4] were used` as well.

Maybe we should add a debug log here to show the new generated ID, just as a visual check when testing.